internal/kms/kes.go

```diff
@@ -416,11 +441,16 @@ func (c *kesClient) Verify(ctx context.Context) []VerifyResult {
 			HTTPClient: c.client.HTTPClient,
 		}
 
-		// 1. Get stats for the KES instance
-		state, err := client.Status(ctx)
-		if err != nil {
-			results = append(results, VerifyResult{Status: "offline", Endpoint: endpoint})
-			continue
+		state := kes.State{}
+		// Permission on /status is not required to validate kes key encryption and decryption
+		if ctx.Value(CtxValidKey{}) != "true" {
+			// 1. Get stats for the KES instance
+			clientState, err := client.Status(ctx)
+			if err != nil {
+				results = append(results, VerifyResult{Status: "offline", Endpoint: endpoint})
+				continue
+			}
+			state = clientState
```

Comment:
This seems not necessary

Reply:
@aead I am passing values thru the context in order to reuse the Verify function already existing- without requiring the user to have the /status permission and without modifying any other functions which in turn call Verify.

Reply:
Yes, but I don't understand why we need to call `Verify` at all. Refer to: https://github.com/minio/minio/pull/19492#discussion_r1562596492

Reply:
Implemented